### ISSUE TITLE
detected fields API

### ISSUE THREAD

**Issue by trevorwhitney** [grafana/loki#12339]
**Is your feature request related to a problem? Please describe.**
Since "detected fields" (ie. `logfmt` or `json` fields in a log line) are only known at query time, users exploring their logs can only get information about fields present in the first 1000 (or log limit) log lines. This often pictures a very limited picture. Since indexed label/values are often more about infrastructures, fields in log lines can more important to users debugging an application who may be less knowledgeable about the infrastructure.

To enable these use cases, as well as better visualizations of fields present in logs, it would be nice to have an API that returned information about detected fields. 

**Describe the solution you'd like**
An API that returns information about _useful_ detected fields, where usefulness is determined by:
* for strings, is the cardinality of values below some configured threshold?
* for "unwrappable" values, what can it be unwrapped into (ie. duration, bytes, etc.)

The response should include the label names, and should be ordered by usefulness (ie. common fields come first)

**Describe alternatives you've considered**
* A time range could be broken up and sampled to get a better representation of a wider time range, however this still wouldn't include information about usefulness with regards to cardinality and unwrap type.

**Additional context**
Future work may include storing this information in an index, but that's not a requirement for this iteration.

**Comment by matyax** [comment]
@trevorwhitney  This was closed as completed, but I don't see any new PR related with it.

**Comment by trevorwhitney** [comment]
yeah, I closed it as it's not planned. the real solution to me seems to pre-parse log lines in store them in dataobj columnar format, in which case would we still need this "usefulness" score, since the API would be exhaustive and complete?
